tensorflow/python/kernel_tests/init_ops_test.py

```diff
@@ -444,5 +444,47 @@ def testDevice(self):
     self.assertDeviceEqual("/job:ps", var.initializer.device)
 
 
+class OrthogonalInitializerTest(tf.test.TestCase):
+
+  def testInitializerIdentical(self):
+    for dtype in [tf.float32, tf.float64]:
+      init1 = tf.orthogonal_initializer(seed=1, dtype=dtype)
+      init2 = tf.orthogonal_initializer(seed=1, dtype=dtype)
+      self.assertTrue(identicaltest(self, init1, init2))
+
+  def testInitializerDifferent(self):
+    for dtype in [tf.float32, tf.float64]:
+      init1 = tf.orthogonal_initializer(seed=1, dtype=dtype)
+      init2 = tf.orthogonal_initializer(seed=2, dtype=dtype)
+      self.assertFalse(identicaltest(self, init1, init2))
+
+  def testDuplicatedInitializer(self):
+    init = tf.orthogonal_initializer()
+    self.assertFalse(duplicated_initializer(self, init, 1))
+
+  def testInvalidDataType(self):
+    self.assertRaises(
+      ValueError,
+      tf.orthogonal_initializer, dtype=tf.string)
+
+  def testGain(self):
+    num = 100
+    for dtype in [tf.float32, tf.float64]:
+      init1 = tf.orthogonal_initializer(seed=1, dtype=dtype)
+      init2 = tf.orthogonal_initializer(gain=3.14, seed=1, dtype=dtype)
+      with self.test_session(graph=tf.Graph()):
+        t1 = init1([num]).eval()
+      with self.test_session(graph=tf.Graph()):
```

Comment:
It's redundant now -- we haven't cleaned all the other cases up though.

use_gpu=True will try to run on GPU if available, and pass otherwise.

We have CI builds that run on GPU and CPU, so they will eventually get tested on both.